RecCalorimeter/src/components/TubeLayerModuleThetaMergedCaloTool.h

```diff
@@ -0,0 +1,59 @@
+#ifndef RECCALORIMETER_TUBELAYERMODULETHETAMERGEDCALOTOOL_H
+#define RECCALORIMETER_TUBELAYERMODULETHETAMERGEDCALOTOOL_H
+
+// from Gaudi
+#include "GaudiKernel/AlgTool.h"
+
+// k4FWCore
+#include "k4Interface/ICalorimeterTool.h"
+
+class IGeoSvc;
+
+/** @class TubeLayerModuleThetaMergedCaloTool Reconstruction/RecCalorimeter/src/components/TubeLayerModuleThetaMergedCaloTool.h
+ *TubeLayerModuleThetaMergedCaloTool.h
+ *
+ *  Tool for geometry-dependent settings of the digitisation.
+ *  It assumes cylindrical geometry (layers) and phi-theta segmentation.
+ *
+ *  @author Anna Zaborowska
+ *  @author Zhibo Wu
+ */
+
+class TubeLayerModuleThetaMergedCaloTool : public AlgTool, virtual public ICalorimeterTool {
+public:
+  TubeLayerModuleThetaMergedCaloTool(const std::string& type, const std::string& name, const IInterface* parent);
+  virtual ~TubeLayerModuleThetaMergedCaloTool() = default;
+  virtual StatusCode initialize() final;
+  virtual StatusCode finalize() final;
+  /** Prepare a map of all existing cells in current geometry.
+   *   Active layers (cylindrical tubes) are looked in the geometry manager by name ('\b activeVolumeName').
+   *   Corresponding bitfield name is given in '\b activeFieldName'.
+   *   If users wants to limit the number of active layers, it is possible by setting '\b activeVolumesNumber'.
+   *   The total number of cells N = n_layer * n_theta * n_phi, where
+   *   n_layer is number of layers (taken from geometry if activeVolumesNumber not set),
+   *   n_theta is number of eta bins in that layer,
+   *   n_phi is number of phi bins (the same for each layer).
+   *   For more explanation please [see reconstruction documentation](@ref md_reconstruction_doc_reccalorimeter).
+   *   @param[out] aCells map of existing cells (and deposited energy, set to 0)
+   *   return Status code.
+   */
+  virtual StatusCode prepareEmptyCells(std::unordered_map<uint64_t, double>& aCells) final;
+
+private:
+  /// Pointer to the geometry service
+  ServiceHandle<IGeoSvc> m_geoSvc;
+  /// Name of the detector readout
+  Gaudi::Property<std::string> m_readoutName{this, "readoutName", ""};
+  /// Name of active volumes
+  Gaudi::Property<std::string> m_activeVolumeName{this, "activeVolumeName", "LAr_sensitive"};
+  /// Name of active layers for sampling calorimeter
+  Gaudi::Property<std::string> m_activeFieldName{this, "activeFieldName", "active_layer"};
+  /// Name of the fields describing the segmented volume
+  Gaudi::Property<std::vector<std::string>> m_fieldNames{this, "fieldNames"};
+  /// Values of the fields describing the segmented volume
+  Gaudi::Property<std::vector<int>> m_fieldValues{this, "fieldValues"};
+  /// Temporary: for use with MergeLayer tool
+  Gaudi::Property<unsigned int> m_activeVolumesNumber{this, "activeVolumesNumber", 0};
```

Comment:
@BrieucF : can we get rid of this?

Reply:
I think so

Reply:
Is it bothering you? It looks like we could also use this if one wants to limit the number of layer in which we apply noise. I indeed do not see a clear use case for this but why to restrict existing functionalities?

Reply:
I’m against bloatware - simpler code is easier to read, understand, develop, maintain and debug. I don’t see a use case for this, and this is a new tool so we don’t need to add the full kitchen sink back just in case. If we want to have nonoise in some layers we can have empty histograms for those.

Reply:
That is also a valid point, agreed

Reply:
Sorry, perhaps I am really inexperienced on this subject. I checked a recently updated tool for reference. But a Gaudi property with the same purpose is also defined there:
https://github.com/zwu0922/k4RecCalorimeter/blob/main/RecFCCeeCalorimeter/src/components/NoiseCaloCellsVsThetaFromFileTool.h#L102

Maybe we can use this function to achieve our goal?
https://github.com/key4hep/k4geo/blob/main/detectorCommon/include/detectorCommon/DetUtils_k4geo.h#L195
If so, could you please point me to any examples on how to use this function?

Thanks in advance

Reply:
The tool will not work without providing number of ECAL layers as input. For the moment, I prefer to leave this Gaudi property here, just as what we do in all other similar tools.

Reply:
Hi @zwu0922 ,
OK, indeed I tried running the tool without specifying the number of layers and the code that should get it directly from the geometry is not working. Let's keep it like that, I will remove the non-working code in a subsequent PR